### ISSUE TITLE
Rendering multiple Subtrees in dependency tree tab of Dependency tracker

### ISSUE THREAD

**Issue by phoenix-aditya** [DependencyTrack/dependency-track#2572]
### Current Behavior

WRT a discussion regarding an issue in cyclonedx-cdxgen https://github.com/CycloneDX/cdxgen/issues/247#issuecomment-1452233031 
Currently Dependency tracker only supports displaying one tree in the dependency tree section.

So for a given project the tree is displayed as
`
             C
           / 
A -> B 
          \
            D
`
But many repositories that are analyzed contain multiple trees, i.e. for example if a repository has front-end and back-end
then the trees would not be connected with each other as they do not share dependencies.

### Proposed Behavior

I propose that there is a feature in dependency track to render multiple trees,
So for example if a project has front-end and back-end or more sub-projects as part of it 

the trees can be rendered as below
`
                         C
                       / 
front-end-> B 
                       \
                         D
back-end -> F -> G -> H

CLI -> A -> B -> C
`
Therefore displaying multiple roots if there are multiple dependency trees in a project.

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this enhancement was already requested

**Comment by nscuro** [comment]
I don't believe multiple roots are supported or intended to be used by CycloneDX (cc @stevespringett).

The way you'd normally handle this is by introducing a "meta" component that forms the root of your graph.

In your example:

```
  C -> C1 -> C2
 / 
A -> B -> B1 -> B2
 \
   D -> D1
```

* `A` would be the meta component. If the project is a monorepo, then the name of the monorepo may be used.
* `C` would be the frontend, including its own dependency graph (`C1`, `C2`).
* `B` would be the backend, including its own dependency graph (`B1`, `B2`).
* `D` would be the CLI, including its own dependency graph (`D1`).

**Comment by rkg-mm** [comment]
@nscuro any idea how the cyclonedx cli handles this when merging BOMs? Does it introduce such meta component?

**Comment by nscuro** [comment]
@rkg-mm The metadata per default is populated based on the first input BOM passed to the `merge` command: https://github.com/CycloneDX/cyclonedx-cli/pull/240

This behavior is used by DT when it merges its SBOM with the services BOM: 

https://github.com/DependencyTrack/dependency-track/blob/93321f8a82f2eee4c2d42bb6ffc664ee54a82b4d/pom.xml#L391-L416

The precondition to this then is of course that the first BOM *has* a metadata component.

The CLI also has flags for `--group`, `--name`, `--version` etc., but as described in the CLI PR I linked, the resulting `metadata.component` will not have a BOM ref, thus leading to an incomplete dependency graph.

**Comment by ta7mid** [comment]
There are cases where some BOM components shouldn’t be (direct or transitive) dependencies of the meta component, such as when the meta component is a container: it’s inaccurate to say that a container image “depends on” a software it contains.  Syft’s developers [share this sentiment](https://github.com/anchore/syft/issues/3071#issuecomment-2292119425), and it seems that the official CycloneDX recommendation for such cases is to use [nested component assemblies](https://github.com/CycloneDX/specification/discussions/659#discussioncomment-13684009) instead of dependency relationships.

When Syft generates a CycloneDX SBOM for a container image, it correctly emits dependency relationships between packages installed by OS package managers for instance.  But it’s very unfortunate that in Dependency-Track’s visualization all of those are lost since the meta component, being a container, rightly doesn’t have a dependency chain leading to any package.  Having those relationships retained in the visualization would lead to significant gains in productivity when locating vulnerable components.

> I don’t believe multiple roots are supported or intended to be used by CycloneDX (cc [@stevespringett](https://github.com/stevespringett)).

I think visualizing multiple trees does not have to constitute a violation of the spec if it’s clear that the purpose of the visualization is to show the dependency relationships among the components without implying that one of those components the meta component being described by the SBOM.  Indeed, there is no “root” here: it’s a directed graph, not a tree, except that one component is a bit special; and the graph may constitute multiple disconnected subgraphs.

Using a tree-like hierarchical visualization for what’s actually a directed disconnected graph is probably a source of confusion and trouble here.  Let’s say we want to visualize one connected subgraph; which node in the subgraph do you pick as the visualization tree’s root for that subgraph?  If the subgraph contains the meta component, the meta component is the most intuitive choice, but otherwise there’s no obvious answer except to choose randomly.  So I think for better user experience, moving to a non-hierarchical representation and retaining all dependency relationships in the visualization would be better.
